src/main/java/org/opentripplanner/routing/algorithm/mapping/RaptorPathToItineraryMapper.java

```diff
@@ -264,6 +286,27 @@ private boolean isFree(EgressPathLeg<T> egressPathLeg) {
     return egressPathLeg.egress().isFree();
   }
 
+  /**
+   * If a routing result transfers at the very same stop, RAPTOR doesn't add a path leg. However,
+   * sometimes we want to create a zero distance leg so a UI can show a transfer. Since it would
+   * be considered backwards-incompatible, this is an opt-in feature.
+   */
+  private Leg createTransferLegAtSameStop(PathLeg<T> previousLeg, PathLeg<T> nextLeg) {
+    var transferStop = Place.forStop(transitLayer.getStopByIndex(previousLeg.toStop()));
+    return StreetLeg
+      .create()
+      .withMode(TraverseMode.WALK)
+      .withStartTime(createZonedDateTime(previousLeg.toTime()))
+      .withEndTime(createZonedDateTime(nextLeg.fromTime()))
+      .withFrom(transferStop)
+      .withTo(transferStop)
+      .withDistanceMeters(0)
+      .withGeneralizedCost(0)
+      .withGeometry(GeometryUtils.makeLineString(transferStop.coordinate, transferStop.coordinate))
```

Comment:
Btw can these "point lines" cause issues?

Reply:
Well, they would show up as a dot. Is that a problem?

Reply:
I was more worried it would cause some libraries to throw some geometry error since dot is not really a line.

Reply:
Well it has two points (but they are identical). Do you have a suggestion what should be returned instead?

Reply:
Unfortunately I dont. I don't have any clue if this more likely to cause issues than alternatives of not returning any geometry or just one point. Although, since this isn't used by clients by default, I don't think this will break anything by accident.